### DISCUSSION TITLE
【gltfio】can't bind the shoe entity to the right skin.Need Helps!!!

### DISCUSSION THREAD

**Discussion by dandingol03** [google/filament#5637] [General]
Hi.
&ensp;I use the interface `attachSkin` in `FilamentAsset` to bind shoe renderable to the mapped bone in character `gltf`. However i still met some problems, and i need some help on how to use this `attachSkin` interface. 
&ensp;I have two gltf resources, one is the [character](https://drive.google.com/file/d/1TOkgs-kfGXi4uInRrwVeF2bYQg-Ab1YY/view?usp=sharing) and one is the [shoe](https://drive.google.com/file/d/1IgFLrPLYHqS_GZnpgXyFN6yoB4n9mL4N/view?usp=sharing). They share the same bone which is named by `Accessories_Shoe_1824051`, and the index of the  matched skin in `character` is 12. I run the code below on android device. I use two `ModelViewer` to load gltf resources, one is `modelViewer` which is used to load `character` and one is `modelViewer2` which is used to load the `shoe` .

**The code:**
```
modelViewer2.asset?.let {
    modelViewer2.resourceLoader.asyncUpdateLoad()
    modelViewer2.populateScene(it)
}
modelViewer.animator?.apply {
    if (animationCount > 0) {
        val elapsedTimeSeconds = (frameTimeNanos - startTime).toDouble() / 1_000_000_000
        applyAnimation(0, elapsedTimeSeconds.toFloat())
    }

    modelViewer2.asset?.getFirstEntityByName("Accessories_Shoe_1824051")?.let { entity ->
        scene.indirectLight
        modelViewer.asset?.attachSkin(12, entity)
    }
    updateBoneMatrices()
}
modelViewer.render(frameTimeNanos)
```

**Describe the bug**
1.The shoe renderable can't be rendered at the right position. You can understand from the picture below:
![Screenshot_20220604-004406_gltf viewer](https://user-images.githubusercontent.com/13013579/171909194-9b74596c-0c25-4409-9120-08f6873bb71b.jpg)
2.it seems like the `shoe` i try to attach is affected by other joint from the [video](https://drive.google.com/file/d/16k8Y4pIRuKLlo5BD0OztzrE10poBcPIl/view?usp=sharing) i record

**To Reproduce**
1. render the two gltf resouces i have mentioned above
2. attach the `shoe` to the skin in `character` which is referenced by the  bone named by `Accessories_Shoe_1824051`

**Expected behavior**
the shoe is rendered at the correct place

**Smartphone (please complete the following information):**
 - Device: [SAMSUNG S20]
 - OS: [e.g. Android 11]

**Comment by romainguy** [comment]
You shouldn't use two `ModelViewer`, for your particular use case you should directly use the `AssetLoader` API. At the very least make sure you share the same `Engine` instance across the two `ModelViewer` instances

**Reply by dandingol03** [reply]
I made theses two `ModelViewer` instances share the same `Engine` ,`View` ,  `Renderer`, `Camera` and `Scene`.

**Comment by pixelflinger** [comment]
moving this to a discussion...

**Comment by dandingol03** [comment]
@prideout hi, i try the `attach-skin` interface in `FilamentAsset`, but i met some problems. Could you please help me to see what went wrong in the code i have mentioned before if it is convenient for you? These `gltf` resources can be accessed by link. Thanks for your time.

**Reply by prideout** [reply]
Looking at your video, my first guess is that the BONE_INDICES vertex attribute in the shoe model are incorrect, which would explain why the shoe is being affected by the character's arm movements.  The attachSkin method does not change any vertex data, it just associates a mesh with a skin.

**Reply by whereisnothing** [reply]
I have met the same problem with you.The new  added entity does not show at the correct position and display abnormally.Has somebody got the solution,please share with me.Thanks very much.

**Reply by dandingol03** [reply]
> I have met the same problem with you.The new added entity does not show at the correct position and display abnormally.Has somebody got the solution,please share with me.Thanks very much.

feel sad~

**Reply by dandingol03** [reply]
@whereisnothing 哥们有新进展吗，我最近也是项目催的紧，你现在解决绑定部件模型位置不对的问题了吗
